samcli/commands/local/lib/sam_api_provider.py

```diff
@@ -139,7 +140,7 @@ def _extract_from_serverless_api(self, logical_id, api_resource, collector):
                                   working_dir=self.cwd)
         swagger = reader.read()
         parser = SwaggerParser(swagger)
-        apis = parser.get_apis()
+        apis = parser.get_apis(stage_name=stage_name, stage_variables=stage_variables)
```

Comment:
Stage name/variable is parsed in the method calling it `_extract_from_serverless_api`. Stage_name and stage_variables had to be passed in to instantiate the Api object in get_apis since we don't separate the Api's by stage name/variables.